pkg/bootstrap/testdata/auth_golden.json

```diff
@@ -23,7 +23,7 @@
     "stats_tags": [
       {
         "tag_name": "cluster_name",
-        "regex": "^cluster\\.((.+?(\\..+?\\.svc\\.cluster\\.local)?)\\.)"
+        "regex": "^cluster\\.(.*?(?=\\.zone\\.[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+\\.[a-zA-Z0-9_]+$|\\.circuit_breakers\\.[[:alnum:]]+\\.[a-zA-Z_]+$|\\.health_check\\.[a-zA-Z_]+$|\\.outlier_detection\\.[a-zA-Z0-9_]+$|\\.ring_hash_lb\\.[a-zA-Z_]+$|\\.maglev_lb\\.[a-zA-Z_]+$|\\.tcp_stats\\.[a-zA-Z_]+$|\\.ssl\\.ciphers\\.|\\.ssl\\.curves\\.|\\.ssl\\.sigalgs\\.|\\.ssl\\.versions\\.|\\.ssl\\.[a-zA-Z_]+$|\\.upstream_[a-z0-9_]+$|\\.membership_[a-z0-9_]+$|\\.retry_or_shadow_abandoned$|\\.config_reload$|\\.update[a-z_]+$|\\.version$|\\.max_host_weight$|\\.bind_errors$|\\.assignment[a-z_]+$|\\.upstream\\.tx\\.quic_connection_close_error_code_[0-9]+$|\\.upstream\\.rx\\.quic_connection_close_error_code_[0-9]+$|\\.upstream\\.tx\\.quic_reset_stream_error_code_[0-9]+$|\\.upstream\\.rx\\.quic_reset_stream_error_code_[0-9]+$|\\.canary\\.[a-zA-Z_0-9]+$|\\.internal\\.[a-zA-Z_0-9]+$|\\.external\\.[a-zA-Z_0-9]+$|\\.lb_[a-z0-9_]+$|\\.original_dst_host_invalid$)\\.)"
```

Comment:
Is this really the best way to do this? Can we discuss with the Envoy community about making this better? I don't think anyone would think this is a reasonable configuration - it also is probably extremely expensive.

Reply:
Yeah, I still think we should be able to do some form of escaping while preserving existing behavior for default k8s service names -- and avoid this massive regex.

Reply:
The metrics are `cluster.<clustername>.<metricname>`. Can we just match "between first and last `.`"? Or does that require lookaheads not in envoy?

Reply:
Hi @howardjohn some metrics have `.` in the name. for example
```
cluster.outbound|443|canary|foo.zufardhiyaulhaq.com.ssl.connection_error
```
`ssl.connection_error` is the name